search/viewport_search_callback.cpp

```diff
@@ -27,7 +27,24 @@ void ViewportSearchCallback::operator()(Results const & results)
     m_hotelsModeSet = true;
   }
 
-  if (!results.IsEndMarker())
+  // We need to clear old results and show a new bunch of results when
+  // the search is completed normally (note that there may be empty
+  // set of results), or when the search is not completed and there is
+  // something in results.
+  //
+  // We don't need to clear old results or show current results if the
+  // search is cancelled, because:
+
```

Comment:
Add "//" here too, please. This is one comment.

Reply:
Done